roadmap.md

```diff
@@ -0,0 +1,187 @@
+Tekton:
+
+- [Failure strategies](https://github.com/tektoncd/pipeline/issues/1684)
+- [Conditional execution](https://github.com/tektoncd/pipeline/blob/master/docs/conditions.md)
+  is nearly where we want it, though feedback has indicated this doens't always
+  work the way users expect it to and could use a bit of additional polish.
+- [Pause and resume](https://github.com/tektoncd/pipeline/issues/233)
+- [Partial execution](https://github.com/tektoncd/pipeline/issues/50)
+- [Notifications](https://github.com/tektoncd/pipeline/issues/1740)
+- [`Task` Versioning](https://github.com/tektoncd/pipeline/issues/1839) - Also required
+  for [Catalog](#catalog) success
+- [Alternative Task Implementations](https://github.com/tektoncd/pipeline/issues/215)
+- [Local Execution](https://github.com/tektoncd/pipeline/issues/235)
+- Testing and debugging frameworks
+- Emiting events throughout `Pipeline` execution
+- [Config as code](https://github.com/tektoncd/pipeline/issues/859) - This has some cross
+  over with [a related Triggers issue](https://github.com/tektoncd/triggers/issues/189)
+- [Performant Tekton](https://github.com/tektoncd/pipeline/issues/540) - We should set
+  performance requirements and measure against them
+- [Adding support for other architectures](https://github.com/tektoncd/pipeline/issues/856)
+- [Rich type support for params](https://github.com/tektoncd/pipeline/issues/1393)
+
+#### Task Interfaces and PipelineResources
+
+In alpha Tekton Pipelines, [`PipelineResources`](https://github.com/tektoncd/pipeline/issues/1673)
+were the interface between `Tasks` in a `Pipeline`: they were used for sharing typed
+data and variables between `Tasks`. This is being revisited and the feature has been
+deconstructed into:
+
+* `workspaces` - A way for a `Task` to declare data it needs and provides without
+  needing to know about the underlying mechanism used
+* `Task results` (aka "output params") - A way for a `Task` to provide values as outputs
+  which can be provided to downstream `Tasks`
+
+After adding these features we will once again reconsider the design of `PipelineResources`
+and decide what additional features, if any, we need.
+
+We will also need to decide how to continue to release and maintain the code that
+is used to build the images used by the existing `PipelineResources`:
+https://github.com/tektoncd/pipeline/tree/master/cmd.
+
+### Triggers
+
+In 2019 we created a simple system for creating instances of Tekton resources, triggered
+by json payloads sent to HTTP endpoints (Tekton Triggers).
+
+In 2020 we would like to add missing features and then push for
+[Triggers to be Beta and later GA](#beta-and-ga):
+
+* [Improved support for many EventListeners](https://github.com/tektoncd/triggers/issues/370)
+* [Pluggable core interceptors](https://github.com/tektoncd/triggers/issues/271)
+* [Support for expressions in TriggerBindings](https://github.com/tektoncd/triggers/issues/271)
```

Comment:
Technically TriggerBindings already support "expressions" -- just the JSONPath kind. The reason for considering CEL was adding support for extensible functions to Bindings..which I guess is my long winded way of saying a better title might be "Support for extensible functions in TriggerBindings" or something :smiley:

Reply:
haha kk :D

Reply:
maybe ill change it to something like "increased expression support"

the thing is these "extensible functions" i think are only extensible afaik in that you can build functions into tekton triggers itself, but it's not like users can extend it with functions?